scripts/zkai_gkr_d128_projection_scaling_preflight_gate.py

```
@@ -0,0 +1,638 @@
+    proof_size_comparable = require_bool(row["proof_size_comparable"], f"{row_id} proof comparable")
+    require_str(row["source_status"], f"{row_id} source status")
+    require_str(row["recommendation"], f"{row_id} recommendation")
+    require_str(row["source_artifact"], f"{row_id} source artifact")
+    if not isinstance(row["non_claims"], list) or not set(NON_CLAIMS).issubset(set(row["non_claims"])):
+        raise GkrProjectionPreflightError(f"{row_id} non-claim inventory drift")
+    if proof_size_comparable:
+        raise GkrProjectionPreflightError(f"{row_id} proof-size comparability overclaim")
+    if row["matched_workload"]:
+        raise GkrProjectionPreflightError(f"{row_id} matched workload overclaim")
+    if object_class.startswith("tiny") and "not d128" not in " ".join(row["non_claims"]):
+        raise GkrProjectionPreflightError(f"{row_id} tiny fixture non-claim missing")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)